homeassistant/components/switch/fritzdect.py

```diff
@@ -147,25 +193,45 @@ def __init__(self, fritz, ain):
         self.has_switch = STATE_UNKNOWN
         self.has_temperature = STATE_UNKNOWN
         self.has_powermeter = STATE_UNKNOWN
+        self.is_online = False
 
     def update(self):
         """Get the latest data from the fritz box."""
-        from requests.exceptions import RequestException
+        if self.is_online is False:
+            _LOGGER.error("Not online skipping request")
+            return
 
         try:
             actor = self.fritz.get_actor_by_ain(self.ain)
+        except (RequestException, HTTPError):
+            _LOGGER.error("Request to actor registry failed")
+            self.state = STATE_UNKNOWN
+            self.temperature = STATE_UNKNOWN
```

Comment:
State attributes should not be reported if they are not known. Set them to `None` here and check for that in the property and only include them if not `None`.

Reply:
See above - the attributes are only reported, if they are != STATE_UNKNOWN. Only "state" is always reported (and in STATE_UNKNOWN initially) as not reporting the state does not seem to be possible.

Reply:
Good, then just replace `STATE_UNKNOWN` with `None` everywhere.